### ISSUE TITLE
Flymake syntax check fails for python3-pylint because of pylint exit status

### ISSUE THREAD

**Issue by landtuna** [jorgenschaefer/elpy#1157]
My elpy-config shows this:
`Syntax checker....: python3-pylint (/usr/bin/python3-pylint)`

When it checks the buffer, I get the error "Flymake: Configuration error has occurred while running (python3-pylint <my-file-path>). Flymake will be switched OFF".

I believe this is happening because when pylint 1.6.4 finds an error, it returns an error code. (When I run it on the offending code, it produces a perfectly nice report with some warnings about missing docstrings and too many local variables but returns to the shell an exit code of 24.)

**Comment by jorgenschaefer** [comment]
Hello, and thanks for the report! An error code is the correct behavior, this is something else. I suspect `python3-pylint` has an output format that flymake does not know how to handle. Can you provide a command line switch to it so it uses the conventional file:line:column:error format?

**Comment by landtuna** [comment]
I think what we're looking for is `python3-pylint -f parseable`. But when I put that in the "Elpy Syntax Check Command" customize entry, it tries to look for an executable named `python3-pylint -f parseable`. Is there a way to set that argument without writing a wrapper script?

**Comment by landtuna** [comment]
Also - I confirmed that it works with a wrapper script

**Comment by jorgenschaefer** [comment]
Hm. Apparently, flymake treats that value as a command, not as a command string with arguments … Annoying.

You could try flycheck over flymake, just disable `elpy-flymake` in `elpy-modules` … maybe that has native support for your checker.

**Comment by landtuna** [comment]
Yes, that's actually what I've been doing - flycheck seems to work pretty well. Thanks for looking into this!

**Comment by jorgenschaefer** [comment]
Ok. I think Elpy really should just switch over to Flycheck … when I find the time. :-|
